qiskit/algorithms/evolvers/classical_methods/scipy_real_evolver.py

```diff
@@ -0,0 +1,247 @@
+from typing import Tuple, List
+import scipy.sparse as sp
+from scipy.sparse.linalg import bicg, norm
+import numpy as np
+
+from qiskit.quantum_info.states import Statevector
+from qiskit import QuantumCircuit
+from qiskit.opflow import StateFn
+
+from .scipy_evolver import SciPyEvolver
+from ..evolution_problem import EvolutionProblem
+from ..evolution_result import EvolutionResult
+from ..real_evolver import RealEvolver
+
+
+class SciPyRealEvolver(RealEvolver, SciPyEvolver):
+    r"""Classical Evolver for real time evolution.
+
+    Evolves an initial state :math:`|\Psi\rangle` for a time :math:`t`
+    under a Hamiltonian  :math:`H`, as provided in the ``evolution_problem``.
+
+    In order to perform one timestep :math:`\Delta t` of the evolution, we need to implement:
+
+    .. math::
+
+        |\Psi ( t + \Delta t ) \rangle = \exp(-i \Delta t H) |\Psi (t ) \rangle
+
+    Which can be approximated as:
+
+    .. math::
+
+        \exp(-i \Delta t H) = \left(1+ i \frac{\Delta t}{2} H \right) ^{-1}
+                                \left( 1 - i \frac{\Delta t}{2} H \right)
+                                + O\left( \Delta t^3 \right)
+
+    Note that this operator is unitary, and thus we won't need to renormalize the state at
+    each step. In order to find :math:`|\Psi ( t + \Delta t ) \rangle` we then
+    need to solve a linear system of equations:
+
+    .. math::
+
+        \left( 1+ i \frac{\Delta t}{2} H \right) |\Psi ( t + \Delta t ) \rangle
+                = \left( 1 - i \frac{\Delta t}{2} H \right) |\Psi( t ) \rangle
+
+    This system of equations will be solved using the Bi-conjugate Gradient method.
+
+    """
+
+    def __init__(
+        self, threshold: float = 1e-3, bicg_err: float = 0.1, max_iterations: int = np.inf
+    ):
+        r"""
+        Args:
+            threshold: The threshold for the error. If timesteps is `None` this will be used
+                to estimate the necessary number of timesteps to reach a threshold error.
+            bicg_err: Needs to be a value between 0 and 1. `bicg_err` will be the percentage of error
+                that comes from solving the linear system of equation with BiCG and
+                the rest will be the error that comes from the Taylor expansion.
+            max_iterations: The maximum number of iterations to perform.
+
+        Raises:
+            ValueError: If `bicg_err` is not between 0 and 1.
+            ValueError: If `threshold` is not positive.
+            ValueError: If `max_iterations` is not a positive integer.
+
+        """
+
+        if max_iterations < 1:
+            raise ValueError("`max_itertations` must be a positive integer.")
+
+        if bicg_err < 0 or bicg_err > 1:
+            raise ValueError("`bicg_err` must be between 0 and 1.")
+
+        if threshold < 0:
+            raise ValueError("`threshold` must be positive.")
+
+        self.max_iterations = max_iterations
+        self.threshold = threshold
+        self.bicg_err = bicg_err
```

Comment:
Similar comment about API ref and see PVQD for Attributes usage to have them in docs